config/base/derive/src/config.rs

```diff
@@ -0,0 +1,477 @@
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use proc_macro_error::abort;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    Data, DataStruct, DeriveInput, Fields, Ident, Lit, LitStr, Meta, Token, Type, TypePath,
+};
+
+use super::utils;
+
+pub(crate) struct EnvPrefix {
+    _ident: Ident,
+    _eq: Token![=],
+    pub(crate) prefix: LitStr,
+}
+
+mod attrs {
+    pub const ENV_PREFIX: &str = "env_prefix";
+    pub const SERDE_AS_STR: &str = "serde_as_str";
+    pub const INNER: &str = "inner";
+}
+
+impl Parse for EnvPrefix {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::ENV_PREFIX)?,
+            _eq: input.parse()?,
+            prefix: input.parse()?,
+        })
+    }
+}
+
+pub(crate) struct Inner {
+    _ident: Ident,
+}
+
+impl Parse for Inner {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::INNER)?,
+        })
+    }
+}
+
+pub(crate) struct SerdeAsStr {
+    _ident: Ident,
+}
+
+impl Parse for SerdeAsStr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::SERDE_AS_STR)?,
+        })
+    }
+}
+
+pub(crate) fn impl_load_env(
+    field_idents: &[&Ident],
+    inner: &[bool],
+    option: &[bool],
+    lvalue: &[proc_macro2::TokenStream],
+    as_str: &[bool],
+    field_ty: &[Type],
+    field_environment: &[String],
+) -> proc_macro2::TokenStream {
+    let set_field = field_ty
+        .iter()
+        .zip(field_idents.iter())
+        .zip(as_str.iter())
+        .zip(lvalue.iter())
+        .zip(option.iter())
+        .map(|((((ty, ident), &as_str_attr), l_value), &is_option)| {
+            let is_string = if let Type::Path(TypePath { path, .. }) = ty {
+                path.is_ident("String")
+            } else {
+                false
+            };
+            // println!("Type in ")
+            // TODO: no need for separate fn
+            let set_field = if is_string {
+                quote! { #l_value = var }
+            } else if as_str_attr {
+                quote! {
+                    #l_value = serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
+                }
+            } else if is_option {
+                quote! {
+                    #l_value = Some(serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?)
+                }
+            }
```

Comment:
This code path should get run when deriving `Configurable` for our `Proxy` structs.